src/inventory.go

```diff
@@ -11,25 +11,33 @@ import (
 	"strings"
 )
 
-func getBinPath() (string, error) {
-	// Check first for RedHat
-	binPath := "/usr/sbin/httpd"
-	_, err := os.Stat(binPath)
-	if err != nil {
-		// If it isn't a RedHat, try with Debian
-		binPath = "/usr/sbin/apache2ctl"
-		_, derr := os.Stat(binPath)
-		if derr != nil {
-			return "", fmt.Errorf("it isn't possible to locate Apache executable")
+func getBinPath(binPath string) (string, error) {
```

Comment:
This whole function is starting to get a bit funky. WDYT about rewriting it, and make it probe from a list and return the first successful entry?

e.g., with some things elided for brevity:

```go
var errBinaryNotFound = errors.New("could not find apache binary")

func getBinPath(binPath string) (string, error) {
	paths := []string{binPath, "/usr/sbin/httpd", "/usr/sbin/apache2ctl"}
	for _, path := range paths {
		_, err := os.Stat(path)
		if err == nil {
			return path, nil
		}
		log.Debugf("Probing for apache binary at %q failed: %v", path, err)
	}

	return "", errBinaryNotFound
}
```

Reply:
In the above snippet we should somewhere handle the argument `binPath` being empty though